### DISCUSSION TITLE
Proxy localhost:port -> subdomain.localhost:another_port doesn't work

### DISCUSSION THREAD

**Discussion by PetrShypila** [chimurai/http-proxy-middleware#588] [Q&A]
### Expected behavior
I have two running dev servers on localhost. I need to proxy a request from `localhost:4000` to `subdomain.localhost:3000`

### Actual behavior
On server launch I receive:
`[HPM] Proxy created: /  ->  http://agent.localhost:3000`
On request I see following message in console:
```[HPM] Error occurred while trying to proxy request /config from localhost:4000 to http://agent.localhost:3000 (ENOTFOUND) (https://nodejs.org/api/errors.html#errors_common_system_errors)```
And in a browser I receive following `504` response:
`Error occurred while trying to proxy to: localhost:4000/config`

At the same time requesting that endpoint in Chrome works fine.
### Setup

* http-proxy-middleware: 0.18.0
* server: express@4.16.2
* webpack-dev-middleware@3.1.0

#### proxy middleware configuration
```javascript
  app.use(require('webpack-dev-middleware')(compiler, {
    noInfo: true,
    publicPath: config.output.publicPath
  }));
  app.use(require('webpack-hot-middleware')(compiler));

  config.devServer.proxyConf.endpoints.forEach((endpoint) => {
    app.use(endpoint, proxy(config.devServer.proxyConf.config));
  });
```
```javascript
  proxyConf: {
    endpoints: [
      "/config",
    ],
    config: {
      target: "http://agent.localhost:3000",
      secure: false,
      changeOrigin: true
    },
  }
```

#### server mounting
```javascript
  app.listen(port, function (err) {
    if(err) {
      console.log(`Error during app launch: ${err.message}`.red);
    } else {
      console.log(`Server is listening on ${port}...`.green);
    }
  });
```

**Comment by simpslandyy** [comment]
Has anyone found a solution to this issue yet?
I'm also getting a ENOTFOUND error, but my target is slightly different, but more or less the same concept.

**Comment by chimurai** [comment]
Care to create a minimal working setup so it can be investigated?

**Comment by PetrShypila** [comment]
Afterall I just started to proxy to another environment. So solution wasn't found

**Comment by dominicfraser** [comment]
Is this helpful? 

https://stackoverflow.com/questions/36745592/node-http-proxy-middleware-not-working-with-local-servers-as-targert

**Comment by a-type** [comment]
I've hit this as well. My path is correct for the target server. I can hit the same endpoint in Chrome (`admin.localhost:8080`), but the proxy middleware will not route it.

Update: I needed to add `admin.localhost` to my hosts file pointed to `127.0.0.1`.

**Comment by camilohollanda** [comment]
putting my local subdomains on hosts file solved the issue to me, not the best solution but ok

**Comment by ihorbond** [comment]
I have a similar issue but my subdomain is dynamic. I tried using router callback so that I can reuse original host (since it includes the subdomain that I need) and just swap out the port but in the end the value from the target options is being used either way. Do you guys have any ideas ?

**Comment by ChristopherJTrent** [comment]
I'm having a similar issue as well.

I've got my frontend and backend being served with separate server processes (they're written in different languages, so as much as I might want to, it's really not possible to integrate them)

Both sides of my application are being proxied using nginx proxy manager.  
My backend is being run at (as an example) `api.foo.bar`, with my frontend at `foo.bar`.  
my configuration is:
```json
{
    "/api": {
        "target": "https://api.foo.bar",
        "pathRewrite": {
            "^/api": ""
        },
        "changeOrigin": true
    }
}
```

I then make a POST request to `/api/auth/validateInvitation` from the frontend, and when I check the console, the request is being made to `https://foo.bar/auth/validateInvitation` instead of `https://api.foo.bar/auth/validateInvitation`, and is subsequently failing because the server hosting the frontend doesn't allow POST requests, to that path or any other.

Is my configuration wrong? am I trying to do something unsupported? Seems a bit weird to me, having an api server running on a subdomain is a pretty common thing.
